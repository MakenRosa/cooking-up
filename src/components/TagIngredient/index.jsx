import { styled } from "styled-components"

const Button = styled.button`
    background: ${props => props.$active ? 'var(--coral, #F0633C)' : 'var(--light-grey, #EFEFEF)'};
    color: ${props => props.$active ? 'var(--cream, #FFFAF3)' : 'var(--grey, #444)'};
    font-weight: ${props => props.$active ? '700' : '400'};
    padding: 8px;
    border-radius: 8px;
    border: none;
    cursor: pointer;
    text-align: center;
    font-family: Nunito Sans;
    font-size: 18px;
    font-style: normal;
    line-height: 150%; 
`

const TagIngredient = ({ ingredient, onClick, active }) => {
    return (<Button $active={active} onClick={onClick}>
        {ingredient.name}
    </Button>)
}

export default TagIngredient