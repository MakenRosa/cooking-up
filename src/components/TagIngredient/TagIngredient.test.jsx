import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import TagIngredient from '.';

describe('TagIngredient', () => {
  const ingredient = { name: 'Tomato' };
  const handleClick = vi.fn();

  it('renders correctly', () => {
    render(<TagIngredient ingredient={ingredient} onClick={handleClick} active={false} />);
    expect(screen.getByText('Tomato')).toBeInTheDocument();
  });

  it('calls onClick when clicked', () => {
    render(<TagIngredient ingredient={ingredient} onClick={handleClick} active={false} />);
    fireEvent.click(screen.getByText('Tomato'));
    expect(handleClick).toHaveBeenCalled();
  });

  it('calls onClick when active is false and button is clicked', () => {
    render(<TagIngredient ingredient={ingredient} onClick={handleClick} active={false} />);
    fireEvent.click(screen.getByText('Tomato'));
    expect(handleClick).toHaveBeenCalled();
  });

  it('changes style when active', () => {
    const { rerender } = render(<TagIngredient ingredient={ingredient} onClick={handleClick} active={false} />);
    expect(screen.getByText('Tomato')).toHaveStyle('background: var(--light-grey, #EFEFEF)');
    
    rerender(<TagIngredient ingredient={ingredient} onClick={handleClick} active={true} />);
    expect(screen.getByText('Tomato')).toHaveStyle('background: var(--coral, #F0633C)');
  });

  it('displays ingredient name', () => {
    render(<TagIngredient ingredient={ingredient} onClick={handleClick} active={false} />);
    expect(screen.getByText('Tomato')).toBeInTheDocument();
  });

  it('matches snapshot when active is false', () => {
    const ingredient = { name: 'Tomato' };
    const { asFragment } = render(<TagIngredient ingredient={ingredient} onClick={() => {}} active={false} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot when active is true', () => {
    const ingredient = { name: 'Tomato' };
    const { asFragment } = render(<TagIngredient ingredient={ingredient} onClick={() => {}} active={true} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
