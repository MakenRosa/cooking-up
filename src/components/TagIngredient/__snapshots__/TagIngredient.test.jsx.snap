// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TagIngredient > matches snapshot when active is false 1`] = `
<DocumentFragment>
  <button
    class="sc-bczSft fmYKzU"
  >
    Tomato
  </button>
</DocumentFragment>
`;

exports[`TagIngredient > matches snapshot when active is true 1`] = `
<DocumentFragment>
  <button
    class="sc-bczSft gwwmbR"
  >
    Tomato
  </button>
</DocumentFragment>
`;
